src/select/__tests__/select.test.js

```diff
@@ -139,74 +125,18 @@ describe('Stateless select', function() {
       expect(instance.state.isDropDownOpen).toBeFalsy();
     });
 
-    describe('On change action', function() {
-      let id = '2';
-      let label = 'label for 2';
-      let textValue = 'some value';
-      const selectedOption = {
-        id,
-        label,
-      };
-      beforeEach(function() {
-        const options = [
-          {
-            id: '1',
-            label: 'label for 1',
-          },
-          {
-            id,
-            label,
-          },
-          {
-            id: '3',
-            label: 'label for 3',
-          },
-          {
-            id: '4',
-            label: 'label for 4',
-          },
-        ];
-        allProps.type = TYPE.search;
-        allProps.options = options;
-        allProps.onChange = jest.fn();
-        allProps.onTextInputChange = jest.fn();
-        wrapper = mount(<Select {...allProps} />);
-      });
-
-      test.each([
-        [
-          STATE_CHANGE_TYPE.select,
-          {option: selectedOption, selectedOptions: [{id, label}]},
-        ],
-        [STATE_CHANGE_TYPE.unselect, {selectedOptions: []}],
-      ])(
-        'should set update of state if change action is %s',
-        (type, expectedResult) => {
-          const e = {
-            target: {
-              value: textValue,
-            },
-          };
-          const event =
-            type === STATE_CHANGE_TYPE.unselect
-              ? [e, undefined, true]
-              : [e, selectedOption];
-          wrapper.instance().onChange(...event);
-          expect(allProps.onChange).toHaveBeenCalledWith(
-            e,
-            Object.assign({}, expectedResult, {type}),
-          );
```

Comment:
This was testing the internal implementation – we should just be testing the public interface of the component.

Reply:
if this is internal , there has to be the way to test it. For instance to emulate `'click` event here. We shoulnd't just remove tests

Reply:
yeah I guess I assumed this behavior was covered by another test that actually simulated click events, but I see now that it was not. i'll add a separate test for that 👍